### PR TITLE
Remove superfluous _validate from Data node

### PR DESCRIPTION
**Pull request by ltalirz**
The Data nodeclass overwrote the `_validate` function but
 (a) did no validation
 (b) included commented code that referenced a private git repository
 (c) returned `None` instead of `True`.

The outcommented code involved validation of the `source` attribute and
the referenced issue pointed out that the value of this attribute was
not officially standardized (potentially leading users to using it in
different ways).

Since there has not been validation of this field since 2015, it seems
safe to assume that it cannot be easily reintroduced at this point.

### CODE REVIEW

File: aiida/orm/nodes/node.py [hunk 1]
```diff
@@ -175,13 +175,15 @@ def initialize(self) -> None:
...
 
     def _validate(self) -> bool:
-        """Check if the attributes and files retrieved from the database are valid.
+        """Validate information stored in Node object.
 
-        Must be able to work even before storing: therefore, use the `get_attr` and similar methods that automatically
-        read either from the DB or from the internal attribute cache.
+        For the :py:class:`~aiida.orm.Node` base class, this check is always valid.
+        Subclasses can override this method to perform additional checks
+        and should usually call ``super()._validate()``!
```
Comment (ramirezfranciscof) [inline aiida/orm/nodes/node.py:182]: Maybe this is obvious if you are used how the `super` works, but I'd still keep this clarification:

```suggestion
        and should usually call ``super()._validate()`` first!
```

or

```suggestion
        and should usually call ``super()._validate()`` at the beginning of it!
```

**Review by ramirezfranciscof — APPROVED** [review]
Hey @ltalirz ! Just one little comment but other than that this looks good to go for me, ping me if you need final approval once you rebase/update.

**Comment by ltalirz** [comment]
thanks, done!

**Comment by ramirezfranciscof** [comment]
Mmm strange i thought that would fix the CI doc problem. Do you understand what is wrong with this? I'll try compiling the docs locally.

File: aiida/orm/nodes/node.py [hunk 2]
```diff
@@ -178,13 +178,15 @@ def initialize(self) -> None:
...
-        Must be able to work even before storing: therefore, use the `get_attr` and similar methods that automatically
-        read either from the DB or from the internal attribute cache.
+        For the :py:class:`~aiida.orm.Node` base class, this check is always valid.
+        Subclasses can override this method to perform additional checks
+        and should usually call ``super()._validate()`` first!
 
-        For the base class, this is always valid. Subclasses will reimplement this.
-        In the subclass, always call the super()._validate() method first!
+        This method is called automatically before storing the node in the DB.
+        Therefore, use :py:meth:`~aiida.orm.entities.Entity.get_attribute()` and similar methods that automatically
```
Comment (ramirezfranciscof) [inline aiida/orm/nodes/node.py:188]: This fixes the problem locally for me.

```suggestion
        Therefore, use :py:meth:`~aiida.orm.entities.EntityAttributesMixin.get_attribute()` and similar methods that automatically
```

**Review by ramirezfranciscof — CHANGES_REQUESTED** [review]

File: aiida/orm/nodes/node.py [hunk 3]
```diff
@@ -178,13 +178,15 @@ def initialize(self) -> None:
...
-        read either from the DB or from the internal attribute cache.
+        For the :py:class:`~aiida.orm.Node` base class, this check is always valid.
+        Subclasses can override this method to perform additional checks
+        and should usually call ``super()._validate()`` first!
 
-        For the base class, this is always valid. Subclasses will reimplement this.
-        In the subclass, always call the super()._validate() method first!
+        This method is called automatically before storing the node in the DB.
+        Therefore, use :py:meth:`~aiida.orm.entities.EntityAttributesMixin.get_attribute()` and similar methods that automatically
+        read either from the DB or from the internal attribute cache.
```
Comment (ramirezfranciscof) [inline aiida/orm/nodes/node.py:189]: Haha damn pyliny. Ok, I hope you don't mind if I just commit this and merge xD.

```suggestion
        Therefore, use :py:meth:`~aiida.orm.entities.EntityAttributesMixin.get_attribute()` and similar methods that
        automatically read either from the DB or from the internal attribute cache.
```

Comment (ramirezfranciscof) [reply aiida/orm/nodes/node.py]: (I changed this because pylint was complaining the line was too long)

**Comment by ltalirz** [comment]
sure, please go ahead!

**Review by ramirezfranciscof — APPROVED** [review]
All good gogogo =P

**Pull request merged**